python/hopsworks/core/flink_cluster_api.py

```diff
@@ -0,0 +1,483 @@
+        headers = {"content-type": "application/json"}
+        _client._send_request(
+            "DELETE", path_params, headers=headers, with_base_path_params=False
+        )
+
+    def stop_job(self, execution, job_id):
+        """Stop specific job of the specific execution of the flink cluster.
+        ```python
+
+        # log in to hopsworks
+        import hopsworks
+        project = hopsworks.login()
+
+        # fetch flink cluster handle
+        flink_cluster_api = project.get_flink_cluster_api()
+        flink_cluster = flink_cluster_api.get_cluster(name="myFlinkCluster")
+
+        # get all executions(This will return empty list of no execution is running on this Flink cluster)
+        executions = flink_job.get_executions()
+
+        # select 1st execution
+        execution = executions[0]
+
+        # stop the job
+        job_id = '113a2af5b724a9b92085dc2d9245e1d6'
+        flink_cluster_api.stop_job(execution, job_id)
+        ```
+
+        # Arguments
+            execution: Execution object.
+            job_id: id of the job within this execution
+        # Raises
+            `RestAPIError`: If unable to stop the job
+        """
+        _client = client.get_instance()
+        path_params = [
+            "hopsworks-api",
+            "flinkmaster",
+            execution.app_id,
+            "jobs",
+            job_id,
+        ]
+        headers = {"content-type": "application/json"}
+        return _client._send_request(
+            "PATCH", path_params, headers=headers, with_base_path_params=False
+        )
+
+    def get_jars(self, execution):
+        """Get already uploaded jars from the specific execution of the flink cluster.
+        ```python
+        # log in to hopsworks
+        import hopsworks
+        project = hopsworks.login()
+
+        # fetch flink cluster handle
+        flink_cluster_api = project.get_flink_cluster_api()
+        flink_cluster = flink_cluster_api.get_cluster(name="myFlinkCluster")
+
+        # get all executions(This will return empty list of no execution is running on this Flink cluster)
+        executions = flink_job.get_executions()
+
+        # select 1st execution
+        execution = executions[0]
+
+        # get jar files from this execution
+        flink_cluster_api.get_jars(execution)
+        ```
+
+        # Arguments
+            execution: Execution object.
+        # Returns
+            `List[Dict]`: The array of dicts with jar metadata.
+        # Raises
+            `RestAPIError`: If unable to get jars from the execution
+        """
+
+        _client = client.get_instance()
+        path_params = ["hopsworks-api", "flinkmaster", execution.app_id, "jars"]
+        headers = {"content-type": "application/json"}
+        response = _client._send_request(
+            "GET", path_params, headers=headers, with_base_path_params=False
+        )
+        return response["files"]
+
+    def upload_jar(self, execution, jar_file):
```

Comment:
Since we can now do FlinkExecution.upload_jar, can we make this and any other function being called from FlinkCluster/FlinkExecution private?